python/dgl/optim/pytorch/sparse_optim.py

```diff
@@ -48,6 +50,9 @@ def __init__(self, params, lr):
                     'MultiGPU rank for each embedding should be same.'
                 assert self._world_size == emb.world_size, \
                     'MultiGPU world_size for each embedding should be same.'
+            name = "{}_{}".format(name, emb.name)
+        self._name = name
+        self._short_name = name[:len(name) if len(name) < 32 else 32]
```

Comment:
Do we gaurantee/require that embeddings have unique names? If not, would a solution be to store a global communicator, and only build a new one if one has not yet been established, thus ensuring we only insert the key once? Essentially how it's done here: https://github.com/dmlc/dgl/blob/0.7.1/python/dgl/nn/pytorch/sparse_emb.py#L117
(Not sure why I didn't do that here previously)

Reply:
Yeah. For DistTensor upon which the distributed.NodeEmbedding is built, the name should be unique.

If a user creates multiple sparse optimizer and we use a global communicator here, is there any racing problems?